hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -148,33 +168,66 @@ class RobotHW : public InterfaceManager
     ERROR
   };
 
-  // Return (in realtime) the state of the last doSwitch()
+  /** \brief Return (in realtime) the state of the last doSwitch() */
   virtual SwitchState switchResult() const
   {
     return DONE;
   }
 
-  // Return (in realtime) the state of the last doSwitch() for a given controller
+  /** \brief Return (in realtime) the state of the last doSwitch() for a given controller */
   virtual SwitchState switchResult(const ControllerInfo& /*controller*/) const
   {
     return DONE;
   }
+  /**\}*/
 
-  /**
-   * Reads data from the robot HW
+  
+  /** \name Control Loop
+   * 
+   */
+   /**\{*/
+
+  /** \brief Read data from the robot hardware.
+   *
+   * The read method is part of the control loop cycle (\ref read, update, \ref write) 
+   * and is used to populate the robot state from the robot's hardware resources
+   * (joints, sensors, actuators). This method should be called before 
+   * controller_manager::ControllerManager::update() and \ref write.
+   * 
+   * \note The name \ref read refers to reading state from the hardware.
+   * This complements \ref write, which refers to writing commands to the hardware.
+   *
+   * Querying WallTime inside \ref read is not realtime safe. The input parameters
+   * \ref time and \ref period make it possible to inject time from any 
+   * realtime backend framework.
    *
    * \param time The current time
    * \param period The time passed since the last call to \ref read
    */
   virtual void read(const ros::Time& /*time*/, const ros::Duration& /*period*/) {}
 
-  /**
-   * Writes data to the robot HW
+  /** \brief Writes commands from controllers to the robot hardware.
+   * 
+   * The write method is part of the control loop cycle (read, update, write) 
+   * and used to send out commands, via the hardware interfaces 
+   * (eg. JointCommandInterface and its subclasses), to your robot's hardware 
+   * esources (joints, actuators). Inside a node that handles the control loop, 
+   * this method should be called after \ref read and controller_manager::ControllerManager::update.
+   * 
+   * \note The name \ref write refers to writing (sending out) data commands,
+   * obtained from controllers, to the robot hardware resources.
+   * Use \ref read to write the state of the robot's hardware resources to data members.
```

Comment:
For consistency with `read()`:

```suggestion
   * \note The name \ref write refers to writing commands to the hardware.
   * This complements \ref read, which refers to reading state from the hardware.
```